cli/src/cmd_api.rs

```diff
@@ -148,19 +149,26 @@ impl crate::AuthenticatedCmd for CmdApi {
 
             // Set this as our body.
             bytes = buf.clone();
+        }
 
-            // Set our params to the query string.
-            if !params.is_empty() {
-                let mut query_string = String::new();
-                for (key, value) in params {
-                    if !query_string.is_empty() {
-                        query_string.push('&');
+        // Set our params to the query string.
+        if !params.is_empty() {
+            let mut query_string = String::new();
+            for (key, value) in params {
+                if !query_string.is_empty() {
+                    query_string.push('&');
+                }
+                match &value {
+                    // Avoid injecting quotes around string values by using the inner string.
+                    Value::String(s) => query_string.push_str(&format!("{}={}", key, s)),
+                    Value::Array(_) | Value::Object(_) => {
+                        bail!("invalid query parameter value {}", value)
                     }
-                    query_string.push_str(&format!("{}={}", key, value));
+                    _ => query_string.push_str(&format!("{}={}", key, value)),
```

Comment:
Small thing, but I would feel better if this match explicitly mentioned them all, otherwise I see this and feel like I have to go look at the serde_json docs to know that this `_` represents `Null | Bool | Number`. Also, what happens if it's null — should we bail in that case too?

https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html

Reply:
I left `Null` as a valid option since it's representable in the query string. I don't think we have any endpoints that would accept it as a valid parameter currently, but we can rely on `nexus` to reject it, and we could conceivably take it as value in the future.

Reply:
What I meant (and should have said) is: if it's `Null`, does that serialize to the string `"null"`, which cannot possibly be something anyone wants from the value `null`?

Reply:
Yes, [that will serialize to "null"](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=56b2ba87d75e47c626196a7c032c6d42).

The user [must pass in the string "null" in the first place](https://github.com/oxidecomputer/oxide.rs/blob/594831e720a0d698377b0a13ef94628b588d80b2/cli/src/cmd_api.rs#L353) for us to receive a `Value::null`, so I think this is reasonable to write to the query string.

Reply:
Added an explicit match.